investors/models.py

```diff
@@ -151,3 +160,67 @@ class Meta:
 
     def __str__(self):
         return f"{self.investor} viewed {self.startup} at {self.viewed_at}"
+
+class FollowedProject(models.Model):
+    """
+    Intermediate model representing a project followed by an investor.
+    Stores additional metadata such as status, notes, and timestamps.
+    """
+    investor = models.ForeignKey(
+        'investors.Investor',
+        on_delete=models.PROTECT,
+        related_name='followed_projects_set',
+        db_column='investor_profile_id',
+    )
+    project = models.ForeignKey(
+        'projects.Project',
+        on_delete=models.PROTECT,
+        related_name='followed_by_investors',
+        db_column='project_id',
+    )
+    followed_at = models.DateTimeField(auto_now_add=True)
+
+    STATUS_CHOICES = [
+        ('watching', 'Watching'),
+        ('interested', 'Interested'),
+        ('contacted', 'Contacted'),
+        ('negotiating', 'Negotiating'),
+        ('passed', 'Passed'),
+    ]
+    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='watching')
+    notes = models.TextField(blank=True, null=True)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def clean(self):
+        if self.notes is None:
+            self.notes = ""
+
+        # Prevent investors from following their own projects
+        inv_user_id = self.investor.user_id if getattr(self, 'investor_id', None) else None
+        proj_startup_user_id = None
+        if getattr(self, 'project_id', None):
+            project = getattr(self, 'project', None)
+            if project and hasattr(project, 'startup') and hasattr(project.startup, 'user_id'):
+                proj_startup_user_id = project.startup.user_id
```

Comment:
The method accesses fields that may not be loaded, risking extra queries or AttributeError. Refactor to ensure related objects are loaded or check existence before access.
```
if self.project_id and hasattr(self, 'project') and getattr(self.project, 'startup', None):
    proj_startup_user_id = getattr(self.project.startup, 'user_id', None)
```